### ISSUE TITLE
gradlew: command not found when using | for no_proxy

### ISSUE THREAD

**Issue by hsmade** [gradle/gradle#15231]
Given:

    GRADLE_OPTS="*.domain1.tld|*.domain2.tld"

Gradlew will fail with:
```bash
/gradlew clean
./gradlew: line 165: *.domain1.tld: command not found
./gradlew: line 165: *.domain2.tld: command not found
Error: Could not find or load main class clean
```
### Current Behavior
I would expect the shell not to parse the `|` here. This seems to work fine for maven and others, but I'm guessing that the `eval set -- ...` in gradlew breaks this.

### Context
What I'm trying to do is to use a proxy except for some domains.

### Steps to Reproduce 
See above

### Your Environment
------------------------------------------------------------
Gradle 4.3
------------------------------------------------------------

Build time:   2017-10-30 15:43:29 UTC
Revision:     c684c202534c4138b51033b52d871939b8d38d72

Groovy:       2.4.12
Ant:          Apache Ant(TM) version 1.9.6 compiled on June 29 2015
JVM:          1.8.0_272 (AdoptOpenJDK 25.272-b10)
OS:           Linux 4.15.0-54-generic amd64

### trace
```bash
$ bash -x ./gradlew clean
+ PRG=./gradlew
+ '[' -h ./gradlew ']'
++ pwd
+ SAVED=/var/lib/jenkins/workspace/seeders/seedmaster
++ dirname ./gradlew
+ cd ./
++ pwd -P
+ APP_HOME=/var/lib/jenkins/workspace/seeders/seedmaster
+ cd /var/lib/jenkins/workspace/seeders/seedmaster
+ APP_NAME=Gradle
++ basename ./gradlew
+ APP_BASE_NAME=gradlew
+ DEFAULT_JVM_OPTS=
+ MAX_FD=maximum
+ cygwin=false
+ msys=false
+ darwin=false
+ nonstop=false
+ case "`uname`" in
++ uname
+ CLASSPATH=/var/lib/jenkins/workspace/seeders/seedmaster/gradle/wrapper/gradle-wrapper.jar
+ '[' -n /var/lib/jenkins/.sdkman/candidates/java/current ']'
+ '[' -x /var/lib/jenkins/.sdkman/candidates/java/current/jre/sh/java ']'
+ JAVACMD=/var/lib/jenkins/.sdkman/candidates/java/current/bin/java
+ '[' '!' -x /var/lib/jenkins/.sdkman/candidates/java/current/bin/java ']'
+ '[' false = false -a false = false -a false = false ']'
++ ulimit -H -n
+ MAX_FD_LIMIT=1048576
+ '[' 0 -eq 0 ']'
+ '[' maximum = maximum -o maximum = max ']'
+ MAX_FD=1048576
+ ulimit -n 1048576
+ '[' 0 -ne 0 ']'
+ false
+ false
++ save clean
++ for i in "$@"
++ printf '%s\n' clean
++ sed 's/'\''/'\''\\'\'''\''/g;1s/^/'\''/;$s/$/'\'' \\/'
++ echo ' '
+ APP_ARGS=''\''clean'\'' \
 '
+ eval set -- -Dhttps.proxyHost=XXXXXXX -Dhttps.proxyPort=3128 '-Dhttps.nonProxyHosts=*.domain1.tld|*.domain2.tld' '"-Dorg.gradle.appname=gradlew"' -classpath '"/var/lib/jenkins/workspace/seeders/seedmaster/gradle/wrapper/gradle-wrapper.jar"' org.gradle.wrapper.GradleWrapperMain ''\''clean'\'' \
 '
++ set -- -Dhttps.proxyHost=XXXXXXXXX -Dhttps.proxyPort=3128 '-Dhttps.nonProxyHosts=*.domain1.tld'
++ '*.domain2.tld' -Dorg.gradle.appname=gradlew -classpath /var/lib/jenkins/workspace/seeders/seedmaster/gradle/wrapper/gradle-wrapper.jar org.gradle.wrapper.GradleWrapperMain clean
./gradlew: line 165: *.domain2.tld: command not found
++ uname
+ '[' Linux = Darwin ']'
+ exec /var/lib/jenkins/.sdkman/candidates/java/current/bin/java clean
Error: Could not find or load main class clean```

**Comment by hsmade** [comment]
Notice that escaping the `\` works:
```bash
$ export GRADLE_OPTS="-Dhttps.proxyHost=XXXXXXXX -Dhttps.proxyPort=3128 -Dhttps.nonProxyHosts=*.domain1.tld\|*.domain2.tld"
jenkins@deploy001:~/workspace/seeders/seedmaster$ ./gradlew clean

BUILD SUCCESSFUL in 0s
1 actionable task: 1 up-to-date
```

But I have no clue if this might break something else, further on

**Comment by mlopatkin** [comment]
This should be fixed in Gradle 7.2 where start scripts were rewritten. Please comment here or open a new issue if the problem persists in 7.2.
